graphix/parameter.py

```diff
@@ -38,70 +37,158 @@ def __init__(self, name):
-        self._assigned = False
+        self._value = None
+        self._assignment = None
+        self._expression = sp.Symbol(name=name)
 
     def __repr__(self):
-        if self._assigned:
-            return f"graphix.Parameter, name = {self.name}, assigned value = {self._value}"
+        if self.assigned:
+            return f"expr = {self._expression}, assignment = [ {self.parameters.pop()} : {self._assignment} ]"
         else:
-            return f"graphix.Parameter, name={self.name}"
-    
+            return f"expr = {self._expression} "
+
     @property
     def name(self):
         return self._name
 
     @property
     def assigned(self):
-        return self._assigned
+        return isinstance(self._assignment, (int, float))
 
-    def bind(self, value, unit="pi"):
-        """Binds the parameters to itself.
-        
-        Parameters
-        ----------
-        values : float
-            value to assign to the parameter.
-        unit : "pi" or "radian"
-            unit of the rotation angle, either in rad/pi or rad.
-        """
-        assert not self._assigned
-        assert isinstance(value, float) or isinstance(value, int)
-        if unit == "pi":
-            self._value = value # rotation angles are in unit of pi for measurement commands
-        elif unit == "radian":
-            self._value = value / np.pi
-        else:
-            raise ValueError(f"Unknown unit {unit}")
-        self._assigned = True
+    @property
+    def expression(self):
+        return self._expression
+
+    @property
+    def value(self):
+        return self._value
+
+    @property
+    def parameters(self):
+        return self._expression.free_symbols
 
-    def bind_from_dict(self, values, unit="pi"):
+    def bind_value(self, value, overwrite=False):
         """Binds the parameters to itself.
-        
+
         Parameters
         ----------
-        values : float
-            value to assign to the parameter.
-        unit : "pi" or "radian"
-            unit of the rotation angle, either in rad/pi or rad.
+        values_dict : float
+            dict of values to assign to the parameter.
+
         """
-        assert not self._assigned
-        assert isinstance(values, dict)
-        if self.name in values.keys():
-            self.bind(values[self._name], unit)
-        else:
-            raise ValueError(f"{self._name} not found")
+
+        # assert isinstance( value_dict , dict)
+        if len(self._expression.free_symbols) == 0:
+            raise ValueError(" No unassinged symbols in self.expression")
+
+        symbol = self._expression.free_symbols.pop()
+        val = self._expression.subs({symbol: value})
+
+        if len(val.free_symbols) == 0:
+
+            if overwrite:
+                self._value = float(val)
+                self._assignment = value
+            else:
+                if not self.assigned:
+                    self._value = float(val)
+                    self._assignment = value
```

Comment:
In my sense, this part is redudant and should be replaced by something more concise. In any case, you are setting `self._value` and `self._assignment`. For me the conditions should only be `if not self.assigned: ... else: raise ValueError("...")`